### ISSUE TITLE
[Bug] Shell - Black Screen (ContentPage) after quick switching tabs

### ISSUE THREAD

**Issue by madrvojt** [xamarin/Xamarin.Forms#6174]
### Steps to Reproduce

1.  Open App
2.  Fast switching tabs
3.  After few click page is Black (Screen Record + screenshot include)

### Basic Information

- Version with issue: Xamarin.Forms  4.0.0.394984-pre10
- IDE:
- Platform Target Frameworks: 9.0 (Pie)
  - Android: 9.0
- Android Support Library Version: 28.0.0.1
- Affected Devices: Samsung SM-J330f (API 26)

### Screenshots
![Screenshot_20190511-152856](https://user-images.githubusercontent.com/11706829/57570389-911e9280-7401-11e9-9379-499a6ea6a413.jpg)

[2019_05_11_15_18_13.mp4.zip](https://github.com/xamarin/Xamarin.Forms/files/3169065/2019_05_11_15_18_13.mp4.zip)

### Reproduction Link

[XamarinForms4Demo.zip](https://github.com/xamarin/Xamarin.Forms/files/3169067/XamarinForms4Demo.zip)

**Comment by kingces95** [comment]
Yep. Jump back and forth and it crashes. Doesn't break :( Prob OOM/Dispose type bug.

**Comment by bytefish** [comment]
I can confirm, that this happens on Android 8.1.0, too. But for me it happens, when I switch between FlyoutItems. I don't see any error in the Device Logs. Maybe it helps: When I move the app to background and open it again, the content is rendered correctly.

**Comment by skordesign** [comment]
I have the same problem. Quick reproduce, just tap 2 tabs icon same time.

**Comment by batfink1234** [comment]
The page loads using pushmodalasync inside the shell instead of pushasync, no navigation bar obviously but just until it gets sorted out.

**Comment by Thanghand** [comment]
Hi, Any updates for this issuse ?

**Comment by PureWeen** [comment]
@madrvojt @bytefish @skordesign @Thanghand 

Are you still seeing this with the latest release of Forms?

**Comment by GeorgeVelikov** [comment]
> @madrvojt @bytefish @skordesign @Thanghand
> 
> Are you still seeing this with the latest release of Forms?

Yes this still happens. I find it only happening when using shell contents in a "Tab". Can't really avoid it if I want to have both bottom and top tab bars

**Comment by MohammadAzhar** [comment]
Hi. I was facing the same issue. It is the issue related to worker thread. you must assign all your properties in to main thread.

`Device.BeginInvokeOnMainThread(() =>
            {// your assignment of properties here});`

**Comment by AdilSaiyad** [comment]
I have also same issue,
any one find solution for this?

**Comment by JFMG** [comment]
I use XF 4.8.0.1687 and can confirm this bug is happened on an iOS 14.4 iPhone 12 simulator on the **top** tab bar. I could not reproduce this on the bottom tab bar.

**Comment by danielmeza** [comment]
I got this issue on XF 5.0.0.2012 and is forcing me to remove the shell and implement the old navigation strategy

**Comment by danielmeza** [comment]
I got this issue on XF 5.0.0.2012 and is forcing me to remove the shell and implement the old navigation strategy

**Comment by danielmeza** [comment]
I got his issue on version XF 5.0.0.2012 and is forcing me to remove the Shell and use the old navigation

**Comment by PureWeen** [comment]
@danielmeza are you seeing this on iOS or Android?

The original issue is for android but seems like the issue has shifted to iOS

**Comment by danielmeza** [comment]
@PureWeen I found the cause of the issue, in my case The issue was reproduced when I change the   `DeviceDisplay.KeepScreenOn` from true to false. I put that login into a Timer

**Comment by Ali-YousefiTelori** [comment]
@danielmeza is right, in android, this code will make this bug, and the screen will going to black:
MainActivity.This.Window.ClearFlags(Android.Views.WindowManagerFlags.KeepScreenOn)
